src/server.js

```diff
@@ -0,0 +1,59 @@
+const express = require('express')
+const fetch = require('node-fetch')
+const cors = require('cors')
+const { execWithPromise, ErrorResponse } = require('./utils')
+const { samRoot } = require('../config')
+
+const MAX_OUTPUT_SIZE = 1024 * 1024 * 10
+
+const app = express()
+app.use(cors())
+app.use('/docs', express.static('docs'))
+
+app.get('/status', async (req, res) => {
+  res.sendStatus(200)
+})
+
+/**
+ * @api {post} /api/convert Convert a notebook to static HTML
+ * @apiName convert
+ * @apiVersion 1.0.0
+ * @apiGroup Convert
+ * @apiParam {String} body Jupyter notebook file
+ * @apiSuccess {String} body HTML representation of notebook
+ */
+app.post('/api/convert', async (req, res) => {
+  try {
+    const authRes = await fetch(
+      `${samRoot}/register/user/v2/self/info`,
+      { headers: { authorization: req.headers.authorization } }
+    ).catch(() => {
+      throw new ErrorResponse(503, 'Unable to contact auth service')
+    })
+    if (authRes.status === 401) {
+      throw new ErrorResponse(401, 'Unauthorized')
+    }
+    if (!authRes.ok) {
+      throw new ErrorResponse(503, 'Failed to query auth service')
+    }
+    const { enabled } = await authRes.json()
+    if (!enabled) {
+      throw new ErrorResponse(403, 'Forbidden')
+    }
+    const { childProcess, promise } = execWithPromise(
+      'jupyter nbconvert --stdin --stdout',
+      { maxBuffer: MAX_OUTPUT_SIZE }
+    )
+    req.pipe(childProcess.stdin)
+    const { stdout } = await promise.catch(error => {
+      throw new ErrorResponse(400, error.toString())
+    })
+    console.log(`Converted ${stdout.length} bytes`) // eslint-disable-line no-console
+    return res.send(stdout)
+  } catch (error) {
```

Comment:
This is a pretty big try block. Worth one more function level?

Reply:
Point taken, but I'd like to leave it for now, given the overall simplicity of the app. If we ever end up with another endpoint, I suspect there will be some obvious abstractions that will shorten things up.